### PR TITLE
Encode + Decode Dates in JSON as ISO 8601

### PR DESCRIPTION
**Pull request by bensyverson**
By default, `JSONEncoder()` encodes & decodes `Date` objects using a timestamp, which is nonstandard in most JSON applications.

This PR updates Vapor to use the standard ISO 8601 format, which will be easier for Vapor API users to process and generate.

### CODE REVIEW

**Comment by bensyverson** [comment]
Ugh, sorry about the whitespace. I'm a tabs man 🤷‍♂️

File: Sources/Vapor/Content/ContentConfig.swift [hunk 1]
```diff
@@ -103,12 +103,32 @@ public struct ContentCoders: Service, ServiceType {
...
+		return formatter
+	}
+	
     public static func `default`() -> ContentConfig {
         var config = ContentConfig()
 
         // json
-        config.use(encoder: JSONEncoder(), for: .json)
-        config.use(decoder: JSONDecoder(), for: .json)
+		let encoder = JSONEncoder()
```
Comment (tanner0101) [inline Sources/Vapor/Content/ContentConfig.swift]: spacing hehe

File: Sources/Vapor/Content/ContentConfig.swift [hunk 2]
```diff
@@ -103,12 +103,32 @@ public struct ContentCoders: Service, ServiceType {
...
+		let decoder = JSONDecoder()
+		
+		if #available(OSX 10.12, *) {
```
Comment (tanner0101) [inline Sources/Vapor/Content/ContentConfig.swift]: Did you try this with Swift 4.1? Or is this a Foundation version related thing 🤔 

Can we provide `iso8601` as an extension where it's _not_ available?

Comment (bensyverson) [reply Sources/Vapor/Content/ContentConfig.swift]: Oh, that's an interesting option! The `#available` check was purely to silence the generated Xcode project, which was complaining about the 10.12 requirement on `iso8601`.

**Comment by tanner0101** [comment]
Went for a simpler solution that just errors instead. Thank you for this PR though!

**Pull request closed without merging**